src/main/java/randoop/condition/ExpectedOutcomeTable.java

```diff
@@ -0,0 +1,156 @@
+import java.util.List;
+import java.util.Set;
+import randoop.test.ExpectedExceptionGenerator;
+import randoop.test.ExtendGenerator;
+import randoop.test.InvalidCheckGenerator;
+import randoop.test.PostConditionCheckGenerator;
+import randoop.test.TestCheckGenerator;
+
+/**
+ * Records the outcome of checking all of the prestate {@link BooleanExpression}s: for the {@link
+ * randoop.condition.specification.Precondition}, the {@link GuardPropertyPair}, and {@link
+ * GuardThrowsPair} for an operation call.
+ *
+ * <p>Conceptually, represents a table, in which each table entry represents the specification for
+ * one declared method. There may be multiple entries in a table, because a method implementation
+ * must satisfy not only the specification written on it, but also any written on method
+ * declarations that it overrides or implements.
+ *
+ * <ol>
+ *   <li>Whether any guard expression for the {@link randoop.condition.specification.Precondition}
+ *       fails, or all are satisfied.
+ *   <li>The set of {@link ThrowsCondition} objects for expected exceptions. An exception is
+ *       expected if the guard of a {@link GuardThrowsPair} is satisfied.
+ *   <li>The expected {@link PropertyExpression}, if any.
+ * </ol>
+ *
+ * <p>For a particular operation call, a table is constructed by calling {@link
+ * OperationConditions#checkPrestate(Object[])} and represents the expected outcome(s) from the
+ * call. The table is used to create a {@link TestCheckGenerator} by calling {@link
+ * #addPostCheckGenerator(TestCheckGenerator)} that, when given to the sequence generator, will
+ * classify the call as follows
+ *
+ * <ol>
+ *   <li>For each table entry with a non-empty expected exception set
+ *       <ul>
+ *         <li>If an exception is thrown by the call and the thrown exception is a member of the
+ *             set, then classify as {@link randoop.main.GenInputsAbstract.BehaviorType#EXPECTED}.
+ *         <li>If an exception is thrown by the call and the thrown exception is not a member of the
+ *             set, classify as {@link randoop.main.GenInputsAbstract.BehaviorType#ERROR} (because
+ *             the specification required an exception to be thrown, but it was not thrown).
+ *         <li>If no exception is thrown, then classify as {@link
+ *             randoop.main.GenInputsAbstract.BehaviorType#ERROR}.
+ *       </ul>
+ *
+ *   <li>If for each table entry, the preconditions failed, classify as {@link
+ *       randoop.main.GenInputsAbstract.BehaviorType#INVALID}.
+ *   <li>For each table entry where all preconditions were satisfied, check the corresponding normal
+ *       post-condition property, if one exists. If any such property fails, then classify as {@link
+ *       randoop.main.GenInputsAbstract.BehaviorType#ERROR}.
+ * </ol>
+ */
+public class ExpectedOutcomeTable {
+
+  /** Indicates whether this table is empty. */
+  private boolean isEmpty = true;
+
+  /** Indicates whether a precondition was satisfied. */
```

Comment:
Document how the rows of the table map to these variables; that is, document the correspondence between the abstraction and the concrete state.